resources/idlharness.js

```diff
@@ -1251,6 +1184,110 @@ IdlInterface.prototype.test_self = function()
 };
 
 //@}
+IdlInterface.prototype.test_immutable_prototype = function(type, obj)
+//@{
+{
+    if (typeof Object.setPrototypeOf !== "function") {
```

Comment:
So I know this isn't introduced by this PR, but the refactoring makes it stand out. This creates bad incentives (i.e. you're better off not implementing `Object.setPrototypeOf` than implementing and not some of what follows.

Can we have a strategy to avoid this?